src/keys.rs

```diff
@@ -23,68 +18,451 @@ pub enum KeyPress {
     Home,
     Insert,
     Left,
-    Meta(char),
     Null,
     PageDown,
     PageUp,
     Right,
-    ShiftDown,
-    ShiftLeft,
-    ShiftRight,
-    ShiftUp,
     Tab, // Ctrl('I')
     Up,
+    #[doc(hidden)]
+    __NonExhaustive,
+}
+
+impl From<char> for Key {
+    #[inline]
+    fn from(c: char) -> Self {
+        Self::Char(c)
+    }
+}
+
+impl Key {
+    #[inline]
+    pub const fn with_mods(self, mods: KeyMods) -> KeyPress {
+        KeyPress::new(self, mods)
+    }
+
+    #[inline]
+    pub const fn shift(self) -> KeyPress {
+        self.with_mods(KeyMods::SHIFT)
+    }
+
+    #[inline]
+    pub const fn ctrl(self) -> KeyPress {
+        self.with_mods(KeyMods::CTRL)
```

Comment:
Hrmm... We can't prevent this without encapsulating stuff more (which would break our ability to `match` on a key press.

The old code had the same problem though -- Nothing stopped a user for looking at `KeyPress::Ctrl('M')`. Events coming out of `rustyline` should always get this right though, but I guess it could confuse people adding bindings.

The only thing I can think is to add some comments warning about it, I think in practice it's pretty unlikely to be hit tho, unless I'm missing something?

Reply:
Actually, I found out that some terminals can provide sequences distinguishing these via CSI u sequences: http://www.leonerd.org.uk/hacks/fixterms/.

So I think it's better as-is? I should make sure these are supported here though.